exoctk/contam_visibility/visibilityPA.py

```diff
@@ -405,17 +405,22 @@ def using_gtvt(
     # (sossContamFig.py)
     # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     badPAs = np.sort(badPAs)
-    grouped_badPAs = [[badPAs[0]]]
 
-    for idx in range(1, len(badPAs)):
+    if len(badPAs > 0):
+        grouped_badPAs = [[badPAs[0]]]
+
+        for idx in range(1, len(badPAs)):
+
+            if ((badPAs[idx - 1] + 1) == badPAs[idx]):
 
-        if ((badPAs[idx - 1] + 1) == badPAs[idx]):
+                grouped_badPAs[len(grouped_badPAs) - 1].append(badPAs[idx])
 
-            grouped_badPAs[len(grouped_badPAs) - 1].append(badPAs[idx])
+            elif ((badPAs[idx - 1] + 1) < badPAs[idx]):
+                grouped_badPAs.append([badPAs[idx]])

```

Comment:
As a list comprehension :)

```python
import itertools
grouped_badPAs = [[e[0] for e in d[1]] for d in itertools.groupby(
                  enumerate(badPAs.tolist()), key=lambda i_x: i_x[0] - i_x[1])]
```

We use [`itertools.groupby`](https://docs.python.org/3/library/itertools.html#itertools.groupby) to do the heavy lifting here. We use enumerate and a lambda function to do the grouping.  When `value - index` (from the enumerate) changes from one item to the next, we have a break. `groupby` returns an iterator of the group index and the group itself as an iterator. We can then iterate over these groups to create the new list of lists.

Reply:
This is great! I think you might have also solved an optimization problem I was having... thanks! I'll be making a future PR implementing `itertools` for list comprehension. @falkben